src/components/cc-popover/cc-popover.js

```diff
@@ -0,0 +1,217 @@
+ *
+ * When the popover is open, pressing `esc` will close the popover and focus the `button` element.
+ *
+ * ## Usage
+ *
+ * ```html
+ * <cc-popover>
+ *   <span slot="button-text">Click me</span>
+ *   <div>This is a content to be displayed when button is clicked.</div>
+  * </cc-popover>
+ * ```
+ *
+ * @cssdisplay block
+ *
+ * @event {CustomEvent} cc-popover:open - Fires whenever the popover is opened.
+ * @event {CustomEvent} cc-popover:close - Fires whenever the popover is closed.
+ *
+ * @slot - The area containing the content of the popover.
+ * @slot button-text - The area containing the button text.
+ *
+ * @cssprop {Number} --cc-popover-z-index - Sets the z-index of the floating content (defaults: `999`).
+ */
+export class CcPopoverComponent extends LitElement {
+  static get properties () {
+    return {
+      accessibleName: { type: String, attribute: 'accessible-name' },
+      hideText: { type: Boolean, attribute: 'hide-text' },
+      icon: { type: Object },
+      isOpen: { type: Boolean, attribute: 'is-open', reflect: true },
+      position: { type: String },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {string|null} Sets the accessibleName property of the underlying `cc-button` element. CAUTION: The accessible name should always start with the visible text if there is one. For instance "add to estimation - NodeJS XS" */
+    this.accessibleName = null;
+
+    /** @type {boolean} Whether the button text should be hidden. */
+    this.hideText = false;
+
+    /** @type {IconModel|null} Sets the button icon. */
+    this.icon = null;
+
+    /** @type {boolean} Whether the popover is opened */
+    this.isOpen = false;
+
+    /** @type {PopoverPosition} Sets the position of the popover relative to the `button` element. */
+    this.position = 'bottom-left';
+
+    /** @type {Ref<HTMLElement>} */
+    this._contentRef = createRef();
+
+    /** @type {Ref<HTMLElement>} */
+    this._buttonRef = createRef();
+
+    this._onOutsideClickHandler = new EventHandler(window, 'click', (event) => {
+      const contentElement = this._contentRef.value;
+      if (contentElement != null && !event.composedPath().includes(contentElement)) {
+        this.close();
+      }
+    });
+    this._onEscapeKeyHandler = new EventHandler(this, 'keydown', (event) => {
+      if (event.key === 'Escape') {
+        this.close();
+      }
+    });
+  }
+
+  /**
+   * Opens the popover.
+   */
+  open () {
+    if (!this.isOpen) {
+      this.isOpen = true;
+      dispatchCustomEvent(this, 'open');
+    }
+  }
+
+  /**
+   * Closes the popover.
+   */
+  close () {
+    if (this.isOpen) {
+      this.isOpen = false;
+      this._buttonRef.value?.focus();
+      dispatchCustomEvent(this, 'close');
+    }
+  }
+
+  /**
+   * Toggle the popover display.
+   */
+  toggle () {
+    if (this.isOpen) {
+      this.close();
+    }
+    else {
+      this.open();
+    }
+  }
+
+  updated (_changedProperties) {
```

Comment:
nitpick: `_changedProperties` does not need to start with a `_` (WebStorm autocomplete I guess)